modules/core/src/shaderlib/misc/uniform-types.ts

```diff
@@ -43,21 +45,21 @@ type NumArray16 = [
 
 type UniformType<ValueT extends number | boolean | number[]> = ValueT extends number | boolean
   ? 'f32' | 'i32' | 'u32'
-  : ValueT extends NumArray2
+  : ValueT extends NumArray2 | Vector2
```

Comment:
Interesting typing... More precise than `NumericArray`.  I have been considering how to type numeric arrays of a specific length, maybe this is the way.

Reply:
Shorthands I've used elsewhere, cribbed from gl-matrix:

https://github.com/donmccurdy/glTF-Transform/blob/52c3e571bdcc79aa447ee492ced4469ce670eebf/packages/core/src/constants.ts#L33-L72

Reply:
The problem that I struggled with is that the math.gl classes like `class Vector2 extends Array<number> {` become typed as `number[]` which doesn't match `[number, number]` which would be the more specific and in many cases more desirable type.

It seemed to me that typing it the way it is done here `Vector2 | [number, number]` could work better in many circumstances.